front/src/applications/operationalStudies/utils.ts

```diff
@@ -209,3 +214,79 @@ export const convertDepartureTimeIntoSec = (departureTime: string) => {
 
 export const isInvalidName = (name?: string | null) =>
   !name || name.length > SMALL_INPUT_MAX_LENGTH;
+
+/**
+ * Check if the scheduled points are honored with a trainSchedule and a trainSummary
+ * @param trainResult
+ * @param trainSummary
+ * @returns true if the scheduled points are not honored
+ */
+export const isScheduledPointsNotHonored = (
+  trainSchedule: TrainScheduleResult,
+  trainSummary: Extract<SimulationSummaryResult, { status: 'success' }>
+): boolean => {
+  if (trainSummary.path_item_times_final.length !== trainSchedule.path.length) {
+    throw new Error('Asserstion failed');
+  }
+  if (!trainSchedule.schedule) return false;
+  return trainSchedule.schedule.some((schedule, index) => {
+    if (schedule.arrival === null) return false;
```

Comment:
You can also test that schedule.arrival is not undefined, and you can remove the cast l237
```suggestion
    if (schedule.arrival === null || schedule.arrival === undefined) return false;
```
then: 
```
sToMs(ISO8601Duration2sec(schedule.arrival))
```

Reply:
Done